src/language-js/embed.js

```diff
@@ -143,19 +143,18 @@ function embed(path, print, textToDoc /*, options */) {
             (parentParent.tag.name === "md" ||
               parentParent.tag.name === "markdown")))
       ) {
-        const doc = textToDoc(
-          // leading whitespaces matter in markdown
-          dedent(parent.quasis[0].value.cooked),
-          {
-            parser: "markdown",
-            __inJsTemplate: true
-          }
-        );
+        const text = parent.quasis[0].value.cooked;
+        const indentation = getIndentation(text);
+        const hasIndent = indentation !== "";
         return concat([
-          indent(
+          (hasIndent ? indent : identity)(
             concat([
               softline,
```

Comment:
I think you need to change this to `literalline` if no indent, see [playground link](https://deploy-preview-3676--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEBLAZgAgBQwE4CucAlJsADpSaYC2AJgAaXwAeMlDA3JQL6XWVBUEABoQEAA4xU0AM7JQAQzx4IAdwAKyhPJSKAbhFR1RIRbJjJ0igDay4YgEZ5FYANZwYAZQmvUUAHNkfCIxOggwK1t7MX97PBgNFwCaRSi7BxAAK1kWACEXd08vRRo4ABl-OHSY8QIYCXqAJhrM3zx45BBHRUcATxtoUwk8fxgAdWMYAAtkAA4ABjERiHtxlwkukbh4-WqxPDgARwJUQ6TFFLSkawyxexpUYMJM2X8AmzgARQIIeFaxDBepM6DNkE1AS5UDZ3gBhCA0VJdKDQfYgAj2AAqvV0t3sPB4QA)